snapcraft/internal/sources.py

```diff
@@ -137,6 +138,29 @@ def __init__(self, source, source_dir, source_tag=None,
             raise IncompatibleOptionsError(
                 'can\'t specify a source-branch for a bzr source')
 
+    def _get_environment(self):
+        # A workaround for LP: #1606203
+        env = os.environ.copy()
+
+        proxies = dict(
+            http_proxy=env.get('http_proxy', ''),
+            https_proxy=env.get('https_proxy', ''),
+        )
+
+        proxies = {key: proxies[key] for key in proxies if proxies[key]}
+
+        for key in proxies:
+            if '@' in proxies[key]:
+                logger.warning(
+                    'Removing the {!r} environment variable as '
+                    'authenticated proxies and bzr will not '
+                    'work'.format(key))
+                proxies[key] = ''
```

Comment:
El 09/09/16 a las 00:22, Leo Arias escribió:
>
> In snapcraft/internal/sources.py 
> <https://github.com/snapcore/snapcraft/pull/781#discussion_r78124084>:
>
> > +        env = os.environ.copy()
> > +
> > +        proxies = dict(
> > +            http_proxy=env.get('http_proxy', ''),
> > +            https_proxy=env.get('https_proxy', ''),
> > +        )
> > +
> > +        proxies = {key: proxies[key] for key in proxies if proxies[key]}
> > +
> > +        for key in proxies:
> > +            if '@' in proxies[key]:
> > +                logger.warning(
> > +                    'Removing the {!r} environment variable as '
> > +                    'authenticated proxies and bzr will not '
> > +                    'work'.format(key))
> > +                proxies[key] = ''
>
> I don't understand why it got so complicated. What is that second 
> assignment you are doing to proxies?
> One simple loop should be enough.
>
> |>>> proxies = dict( ... http_proxy='test@test', ... 
> https_proxy='test@test', ... ) >>> for key in proxies.keys(): ... if 
> '@' in proxies[key]: ... proxies[key] = '' ... >>> proxies 
> {'https_proxy': '', 'http_proxy': ''}|

|What if one of those entries don't exist?|||